### PR TITLE
Implement get_closest_waypoint function

### PR DESCRIPTION
**Pull request by SharkyZg**

### CODE REVIEW

File: ros/src/tl_detector/tl_detector.py [hunk 1]
```diff
@@ -113,27 +113,29 @@ def get_closest_waypoint(self, pose):
...
 
         """
-        
+
         # Very high value is set as as initial distance.
         closest_waypoint_dist = 100000
         closest_waypoint_ind = -1
 
         # Looping through base waypoints to find the one closest to the car.
         for i in range(0, len(self.waypoints.waypoints)):
```
Comment (carlosgalvezp) [inline ros/src/tl_detector/tl_detector.py]: Another cool approach is:

for i, waypoint in enumerate(self.waypoints.waypoints):

Then you don't have to type self.waypoints.waypoints[i] later :)

File: ros/src/tl_detector/tl_detector.py [hunk 2]
```diff
@@ -113,27 +113,29 @@ def get_closest_waypoint(self, pose):
...
             if waypoint_distance < closest_waypoint_dist:
-                # In case that closer waypoint has been found, set new distance and new closest waypoint index.
+                # In case that closer waypoint has been found,
+                # set new distance and new closest waypoint index.
                 closest_waypoint_dist = waypoint_distance
                 closest_waypoint_ind = i
 
-        # It is irrelevant if the closest waypoint is in front or behind the car 
-        # because we will need to search for traffic lights x meters in front of the car anyway. 
+        # It is irrelevant if the closest waypoint is in front or
```
Comment (carlosgalvezp) [inline ros/src/tl_detector/tl_detector.py]: Not 100% sure about this, I will read through the code tomorrow!

Comment (SharkyZg) [reply ros/src/tl_detector/tl_detector.py]: Ok, try to print closest waypoints, then you can se how dens they are, I think that one waypoint doesn't make any difference.

File: ros/src/tl_detector/tl_detector.py (hunk 1 above)
Comment (SharkyZg) [reply ros/src/tl_detector/tl_detector.py]: Yes, thanks for the tip, I'm switching too many programming languages :-)

File: ros/src/tl_detector/tl_detector.py (hunk 2 above)
Comment (carlosgalvezp) [reply ros/src/tl_detector/tl_detector.py]: That's true! I still need to understand better the purpose of this function in the rest of the code
(car_position = self.get_closest_waypoint(self.pose.pose)) so I'll approve this for now :)

**Review by carlosgalvezp — APPROVED** [review]
Good work!

**Pull request merged**